specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json

```diff
@@ -0,0 +1,847 @@
+          "$ref": "#/definitions/SearchResults",
+          "description": "Parameters of the search job that initiated this table."
+        },
+        "restoredLogs": {
+          "$ref": "#/definitions/RestoredLogs",
+          "description": "Parameters of the restore operation that initiated this table."
+        },
+        "resultStatistics": {
+          "readOnly": true,
+          "$ref": "#/definitions/ResultStatistics",
+          "description": "Search job execution statistics."
+        },
+        "plan": {
+          "type": "string",
+          "description": "Instruct the system how to handle and charge the logs ingested to this table.",
+          "enum": [
+            "Basic",
+            "Analytics"
+          ],
+          "x-ms-enum": {
+            "name": "TablePlanEnum",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Basic",
+                "description": "Logs  that are adjusted to support high volume low value verbose logs."
+              },
+              {
+                "value": "Analytics",
+                "description": "Logs  that allow monitoring and analytics."
+              }
+            ]
+          }
+        },
+        "lastPlanModifiedDate": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The timestamp that table plan was last modified (UTC)."
+        },
+        "schema": {
+          "$ref": "#/definitions/Schema",
+          "description": "Table schema."
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Table's current provisioning state. If set to 'updating', indicates a resource lock due to ongoing operation, forbidding any update to the table until the ongoing operation is concluded.",
+          "enum": [
+            "Updating",
+            "InProgress",
+            "Succeeded",
+            "Deleting"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningStateEnum",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Updating",
+                "description": "Table schema is still being built and updated, table is currently locked for any changes till the procedure is done."
+              },
+              {
+                "value": "InProgress",
+                "description": "Table schema is stable and without changes, table data is being updated."
+              },
+              {
+                "value": "Succeeded",
+                "description": "Table state is stable and without changes, table is unlocked and open for new updates."
+              }
+            ]
+          }
+        },
+        "retentionInDaysAsDefault": {
+          "readOnly": true,
+          "type": "boolean",
+          "description": "True - Value originates from workspace retention in days, False - Customer specific."
+        },
```

Comment:
Consider using enums as explained here :

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

Reply:
Server side is serializing a Boolean, can we be sure that client side which generates his client from the swagger will read the response correctly? Boolean in json will just return -> true instead of "True".

I think we should depict the server-side contract as it is.

Reply:
As long as you set modelasstring property to true the client will have no trouble with this. modelling values as enums is pretty standard practise across azure services and is a pretty strong recommendation.

Reply:
This is not a blocking comment. but strongly recommended just as an FYI

Reply:
Made the change.